code/modules/paperwork/photography.dm

```diff
@@ -175,119 +175,141 @@
 	res.Scale(size*32, size*32)
 	// Initialize the photograph to black.
 	res.Blend("#000", ICON_OVERLAY)
+	CHECK_TICK
 
-	var/atoms[] = list()
-	for(var/turf/the_turf in turfs)
+	var/list/atoms = list()
+	for(var/turf/the_turf as anything in turfs)
 		// Add outselves to the list of stuff to draw
 		atoms.Add(the_turf);
 		// As well as anything that isn't invisible.
-		for(var/atom/A in the_turf)
-			if(A.invisibility) continue
-			atoms.Add(A)
+		for(var/atom/cur_atom in the_turf)
```

Comment:
is /atom typechecking here necessary? turf contents shouldn’t have any non-atoms

Reply:
I wasn't sure, but if you think so I'll add that.